### PR TITLE
Add SeasonSelect component for matches view

### PR DESCRIPTION
**Pull request by healfy**
### Summary of changes

Creates a select-season component, and then uses that component for the Matches query.

### CODE REVIEW

File: src/components/common/SeasonSelect.vue [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+<script lang="ts">
+import { computed, ComputedRef, defineComponent } from "vue";
+import { Season } from "@/store/ranking/types";
+import { useRankingStore } from "@/store/ranking/store";
+import { useMatchStore } from "@/store/match/store";
+
+export default defineComponent({
+  name: "SeasonSelect",
+  props: {},
```
Comment (gustav87) [inline src/components/common/SeasonSelect.vue:36]: Delete empty props property.

File: src/store/types.ts [hunk 2]
```diff
@@ -216,3 +216,7 @@ export enum EComputer {
   NORMAL = 1,
   INSANE = 2,
 }
+
+export type Season = {
+  id: number;
+};
```
Comment (gustav87) [inline src/store/types.ts:222]:
```suggestion
```

This type already exists in `src\store\ranking\types.ts`

File: src/components/common/SeasonSelect.vue (hunk 1 above)
Comment (gustav87) [inline src/components/common/SeasonSelect.vue:36]:
```suggestion
```
Empty props not needed. The linter complains about that when upgrading to Vue 3, so might as well delete it now.

File: src/store/types.ts (hunk 2 above)
Comment (healfy) [reply src/store/types.ts]: fixed

File: src/components/common/SeasonSelect.vue (hunk 1 above)
Comment (healfy) [reply src/components/common/SeasonSelect.vue]: fixed

**Pull request merged**